oauth2_client/models.py

```diff
@@ -1,54 +1,259 @@
 """
 Ouath2 client models.
 """
-from django.contrib.postgres.fields import JSONField
 from django.db import models
+from django.forms import model_to_dict
 from django.utils import timezone
+from django.conf import settings
+from datetime import timedelta
+
+from django.utils.translation import ugettext_lazy as _
+from django.contrib.postgres.fields import JSONField
+
+from labsterutils.obj import get_value_or_default
+from oauth2_client.exceptions import TokenError
 
 
 class Application(models.Model):
     """
-    Client credentials for the service provider.
+    Client Application model
     """
-    name = models.CharField(max_length=50, help_text="Example: licence")
-    # base definitions copied from oauth2_provider.Application
-    client_id = models.CharField(max_length=100)
-    client_secret = models.CharField(max_length=100)
+    CLIENT_CONFIDENTIAL = "confidential"
+    CLIENT_PUBLIC = "public"
+    CLIENT_TYPES = (
+        (CLIENT_CONFIDENTIAL, _("Confidential")),
+        (CLIENT_PUBLIC, _("Public")),
+    )
+
+    GRANT_AUTHORIZATION_CODE = "authorization-code"
+    GRANT_IMPLICIT = "implicit"
+    GRANT_PASSWORD = "password"
+    GRANT_CLIENT_CREDENTIALS = "client-credentials"
+    GRANT_JWT_BEARER = "jwt-bearer"
+    GRANT_TYPES = (
+        (GRANT_AUTHORIZATION_CODE, _("Authorization code")),
+        (GRANT_IMPLICIT, _("Implicit")),
+        (GRANT_PASSWORD, _("Resource owner password-based")),
+        (GRANT_CLIENT_CREDENTIALS, _("Client credentials")),
+        (GRANT_JWT_BEARER, _("JWT bearer")),
+    )
```

Comment:
Are all those flows are currently supported by Oauth2Client and are tested?

Reply:
Nope. Removing from the model to avoid confusion.